src/jit/codegen.h

```diff
@@ -317,6 +317,11 @@ class CodeGen : public CodeGenInterface
 
 #if defined(_TARGET_ARM_)
 
+    bool genInstrWithConstant(
+        instruction ins, emitAttr attr, regNumber reg1, regNumber reg2, ssize_t imm, insFlags flags, regNumber tmpReg);
```

Comment:
It does the same thing as an arm64 version, but in that case we actually need to know if we're generating unwind codes or not, which doesn't seem like it should be in the emitter. Also, both of them will possibly use an additional register. In some cases for arm32, mostly for loads/stores of locals, the emitter functions automatically use the reserved register, and pre-codegen is required to allocate the reserved register if necessary. That seems pretty tenuous; it seems better to pass in a temp register if it might be needed. It's possible the related emitter functions should take one. But it kind of seems like this is "higher level", and the emitter functions should generate exactly one instruction.

Reply:
OK, thanks for the explanation